src/components/ButtonHyperlink/ButtonHyperlink.types.ts

```diff
@@ -1,7 +1,10 @@
 import { ReactNode } from 'react';
 import { AriaButtonProps } from '@react-types/button';
+import { AriaLinkProps } from '@react-types/link';
+import { roleType } from './ButtonHyperlink.constants';
 
-export interface Props extends AriaButtonProps<'a'> {
+type myProps = AriaLinkProps & AriaButtonProps<'a'>;
+export interface Props extends myProps {
```

Comment:
you can extends multiple interfaces, no need for the extra type

```suggestion
export interface Props extends AiaLinkProps, AriaButtonProps<'a'> {
```